cpp/src/arrow/gpu/cuda_memory.cc

```diff
@@ -501,5 +502,27 @@ Result<std::shared_ptr<MemoryManager>> DefaultMemoryMapper(ArrowDeviceType devic
   }
 }
 
+Result<std::shared_ptr<MemoryManager>> DefaultGPUMemoryMapper(int64_t device_id) {
+  ARROW_ASSIGN_OR_RAISE(auto device, arrow::cuda::CudaDevice::Make(device_id));
+  return device->default_memory_manager();
+}
```

Comment:
We should probably include some sort of customizations on the cuda device to ensure it uses the appropriate allocation type (HOST/MANAGED/etc).

Reply:
I somewhat blindly copied this from an existing function:

https://github.com/apache/arrow/blob/1781b3248743812f737f32c244434b00e8d18579/cpp/src/arrow/gpu/cuda_memory.cc#L488-L497

But yes, that just ignores the device type at the moment. Looking at the code, it seems that CudaDevice currently only supports CUDA allocation type?

Then for this PR I can just remove the other two?

Reply:
That seems reasonable for now. We can update and handle the other types in a future update